zanata-war/src/main/java/org/zanata/service/impl/UserAccountServiceImpl.java

```diff
@@ -89,10 +89,9 @@ public HAccountResetPasswordKey requestPasswordReset(
             + account.getPerson().getName() + System.currentTimeMillis()));
 
         account.setAccountResetPasswordKey(key);
-        accountResetPasswordKeyDAO.makePersistent(key);
+        key = accountResetPasswordKeyDAO.makePersistent(key);
+        accountResetPasswordKeyDAO.flush();
```

Comment:
So either flush or `@Transactional` annotation? I think its best to have both. What do you think?